frontend/ui/src/app/licenses/edit-license.component.ts

```diff
@@ -0,0 +1,267 @@
+import {
+  AfterViewInit,
+  Component,
+  effect,
+  ElementRef,
+  forwardRef,
+  inject,
+  Injector,
+  OnDestroy,
+  OnInit,
+  signal,
+  ViewChild,
+  WritableSignal,
+} from '@angular/core';
+import {AsyncPipe} from '@angular/common';
+import {AutotrimDirective} from '../directives/autotrim.directive';
+import {
+  ControlValueAccessor,
+  FormArray,
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  NG_VALUE_ACCESSOR,
+  NgControl,
+  ReactiveFormsModule,
+  TouchedChangeEvent,
+  Validators,
+} from '@angular/forms';
+import {faChevronDown, faMagnifyingGlass, faPen, faPlus, faXmark} from '@fortawesome/free-solid-svg-icons';
+import {first, firstValueFrom, map, Subject, switchMap, takeUntil} from 'rxjs';
+import {ApplicationLicense} from '../types/application-license';
+import {ApplicationsService} from '../services/applications.service';
+import {Application, ApplicationVersion} from '@glasskube/distr-sdk';
+import {UsersService} from '../services/users.service';
+import dayjs from 'dayjs';
+import {CdkConnectedOverlay, CdkOverlayOrigin} from '@angular/cdk/overlay';
+import {dropdownAnimation} from '../animations/dropdown';
+import {FaIconComponent} from '@fortawesome/angular-fontawesome';
+
+@Component({
+  selector: 'app-edit-license',
+  templateUrl: './edit-license.component.html',
+  imports: [AsyncPipe, AutotrimDirective, ReactiveFormsModule, CdkOverlayOrigin, CdkConnectedOverlay, FaIconComponent],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => EditLicenseComponent),
+      multi: true,
+    },
+  ],
+  animations: [dropdownAnimation],
+})
+export class EditLicenseComponent implements OnInit, OnDestroy, AfterViewInit, ControlValueAccessor {
+  private injector = inject(Injector);
+  private readonly destroyed$ = new Subject<void>();
+  private readonly applicationsService = inject(ApplicationsService);
+  private readonly usersService = inject(UsersService);
+  applications$ = this.applicationsService.list();
+  customers$ = this.usersService.getUsers().pipe(
+    map((accounts) => accounts.filter((a) => a.userRole === 'customer')),
+    first()
+  );
+
+  private fb = inject(FormBuilder);
```

Comment:
The form builder is only used for one field here, is this necessary? I think it would be better to use it for all fields or not at all